packages/valory/skills/abstract_round_abci/abci_app_chain.py

```diff
@@ -184,13 +184,18 @@ def get_paths(
             )
             # we now check that the pre conditions of the next app
             # are compatible with the post conditions of the current apps.
-            diff = set.difference(
-                set(next_app.db_pre_conditions[next_initial_state]),
-                accumulated_post_conditions,
-            )
-            if len(diff) != 0:
-                raise ValueError(
-                    f"Pre conditions '{diff}' of app '{next_app}' not a post condition of app '{current_app}' or any preceding app in path {path}."
+            if next_initial_state in next_app.db_pre_conditions:
+                diff = set.difference(
+                    set(next_app.db_pre_conditions[next_initial_state]),
+                    accumulated_post_conditions,
+                )
+                if len(diff) != 0:
+                    raise ValueError(
+                        f"Pre conditions '{diff}' of app '{next_app}' not a post condition of app '{current_app}' or any preceding app in path {path}."
+                    )
+            else:
+                _default_logger.warning(
+                    f"No pre-conditions have been set for {next_initial_state}!"
```

Comment:
Otherwise, we were getting a non-informative `KeyError`, e.g., [[0.6.0] feat: add pre- and post- conditions on db at fsm level #6358](https://github.com/valory-xyz/open-autonomy/actions/runs/3703352842/jobs/6274692428). I updated the code so that the dev can choose not to use any pre-conditions, e.g., they might be empty. We now just log a warning about it, as I thought it was more appropriate than raising.

Reply:
It should not be allowed to have no pre_conditions set for all initial states! They should explicitly set empty preconditions.

Reply:
Then let's convert the warning with a `ValueError` and the message `f"No pre-conditions have been set for {next_initial_state}!"`. Otherwise, the `KeyError` is confusing.

Reply:
Done in b70515b4968d78a13358c311c7e43f9e82868a2e.